fe/fe-core/src/main/java/com/starrocks/connector/iceberg/IcebergAlterTableExecutor.java

```diff
@@ -495,4 +531,110 @@ private void expireSnapshots(List<ConstantOperator> args) {
             expireSnapshots.commit();
         });
     }
+
+    private void removeOrphanFiles(List<ConstantOperator> args) {
+        if (args.size() > 1) {
+            throw new StarRocksConnectorException("invalid args. only support " +
+                    "`older_than` in the remove orphan files operation");
+        }
+
+        long olderThanMillis;
+        if (args.isEmpty()) {
+            LocalDateTime time = LocalDateTime.now(TimeUtils.getTimeZone().toZoneId());
+            olderThanMillis = time.minus(DEFAULT_RETENTION_THRESHOLD).toInstant(ZoneOffset.UTC).toEpochMilli();
+        } else {
+            LocalDateTime time = Optional.ofNullable(args.get(0))
+                    .flatMap(arg -> arg.castTo(Type.DATETIME)
+                    .map(ConstantOperator::getDatetime))
+                    .orElseThrow(() -> new StarRocksConnectorException("invalid arg %s", args.get(0)));
+            olderThanMillis = Duration.ofSeconds(time.atZone(TimeUtils.getTimeZone().toZoneId()).toEpochSecond()).toMillis();
+        }
+
+        if (table.currentSnapshot() == null) {
```

Comment:
I think this is useless, if I drop a iceberg table, the metadata/data files would still exists, and I recreate a iceberg table using the same location, the new create iceberg table snapshot could be null, then run removeOrphanFiles should remove the metadata/data which belone to the old iceberg table

Reply:
This is a valid concern. When a table is dropped and recreated using the same location, there's a potential issue with orphan file cleanup procedures. The procedure needs to correctly identify which files are truly orphaned versus which ones should be part of the newly created table.
Link to trino about this behavior discuss https://github.com/trinodb/trino/issues/12573

Reply:
Also, I've noticed that Starrocks drop table default behavior won't delete phyical file. If user create new snapshot and we remove orphan file. maybe those behavior will be conflict?

Reply:
Yes, starrocks drop table xxx can not delete the file in file system, you can use `delete table xxx force` to delete them

Reply:
It's a good idea to prevent create iceberg table on non-empty location, we can support this later

Reply:
If you can, please assign it to me.

Reply:
you can create an issue, and assign to yourself